packages/api/src/beacon/server/events.ts

```diff
@@ -49,7 +53,12 @@ export function getRoutes(config: ChainForkConfig, api: ServerApi<Api>): ServerR
             // The client may disconnect and we need to clean the subscriptions.
             req.raw.once("close", () => resolve());
             req.raw.once("end", () => resolve());
-            req.raw.once("error", (err) => reject(err));
+            req.raw.once("error", (err) => {
+              if ((err as unknown as {code: string}).code === "ECONNRESET") {
```

Comment:
I review our code and did some research on how `EventSource` behaves. It looks like we are closing this correctly on the client by calling `.close()`. 

It is invoking both the `close` and `error` event on the server, this is a bit strange to me but it looks like that's how `EventSource` works, it doesn't gracefully close for some reason.

Another thing is that usually this would be handled by fastify which also just ignore errors with code `ECONNRESET`, see https://github.com/fastify/fastify/blob/cc347d7c0b4266097b61b126158b797878668353/fastify.js#L667C23-L667C33

But since the event API is not handled by fastify we need to do it ourselves and applying same pattern sounds good to me.

We might even consider just resolving here instead of rejecting with abort error but I guess if this error is not logged due to it being ignored upstream this is also fine.

Reply:
I feel we should abort here on `ECONNRESET` and let the higher level stack handle that error, as it's currently doing. Once we have more resources to knew if the reset was actually abort or gracefully closed by client then we can decide to resolve the later scenario.

Reply:
Yeah, sounds good to me 👍

Reply:
> It is invoking both the close and error event on the server, this is a bit strange to me but it looks like that's how EventSource works, it doesn't gracefully close for some reason.

This might be related to
- https://github.com/nodejs/node/issues/42154

Reply:
This is also simliar to issue reported here
- https://github.com/websockets/ws/issues/1256